rpc/src/v1/impls/eth.rs

```diff
@@ -451,11 +456,38 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> EthClient<C, SN, S
 				self.miner
 					.pending_state(info.best_block_number)
 					.map(|s| Box::new(s) as Box<dyn StateInfo>)
-					.unwrap_or(Box::new(self.client.latest_state()) as Box<dyn StateInfo>)
+					.unwrap_or_else(|| {
+						// todo[dvdplm]: this strikes me as odd. If I call `get_state` with
```

Comment:
Well, that's exactly why it's `StateOrBlock`. Afair we can't use `Block` as it's only for states that are in the db already. Similarly we can use `State` for past blocks, as it's not exposed from the client (cause it's not needed).

Reply:
My `todo` is unclear, apologies. My point is: if I call `get_state(BlockNumber::Latest)` I get `StateOrBlock::Block` but if I call `get_state(BlockNumber::Pending)` and I'm not mining so I hit the fallback path then I'll get a `StateOrBlock::State` back. 
It felt a bit inconsistent, is all.

Reply:
hmm, right. But do you feel it breaks something?

Reply:
No, I don't think it's a bona fide bug, just a wart.